core/src/main/scala/org/broadinstitute/dsde/rawls/monitor/AvroUpsertMonitor.scala

```diff
@@ -157,22 +189,15 @@ class AvroUpsertMonitorActor(
     case StartMonitorPass =>
       // start the process by pulling a message and sending it back to self
       pubSubDao.pullMessages(pubSubSubscriptionName, 1).map(_.headOption) pipeTo self
+      arrowPubSubDAO.pullMessages(arrowPubSubSubscriptionName, 1).map(_.headOption) pipeTo self
 
     case Some(message: PubSubMessage) =>
-      //we received a message, so we will parse it and try to upsert it
-      logger.info(s"received avro upsert message: $message")
-      val (jobId, file) = parseMessage(message)
-
-      file match {
-        case "upsert.json" =>
-          logger.info(s"processing $file message for job $jobId with ackId ${message.ackId} in message: $message")
-          toFutureTry(initAvroUpsert(jobId, message.ackId)) map {
-            case Success(_) => writeResult(s"$jobId/success.json", resultString("import successful")).map(_ => ImportComplete) pipeTo self
-            case Failure(t) =>  writeResult(s"$jobId/error.json", resultString(t.getMessage)).map(_ => ImportComplete) pipeTo self
-          }
-        case _ =>
-          logger.info(s"found ignorable file $file message for job $jobId with ackId ${message.ackId} in message: $message")
-          acknowledgeMessage(message.ackId).map(_ => ImportComplete) pipeTo self //some other file (i.e. success/failure log, metadata.json)
+      // we received a message, so we will parse it and try to upsert it
+      logger.info(s"received async upsert message: $message")
+      if (message.attributes.contains("jobId")) {
+        importEntities(message)
+      } else {
+        arrowImportEntities(message)  // remove when cutting over to import service
```

Comment:
We decide whether to go down the CF path or the import service path based on whether the attributes contain a jobId. The CF solution has the metadata in the message body, while the import service has the metadata as attributes on the message